manifests/init.pp

```diff
@@ -6,18 +6,118 @@
+  Optional[String]                                $pe_username      = undef,
+  Optional[Sensitive[String]]                     $pe_password      = undef,
+  Optional[String]                                $pe_token         = undef,
+  Optional[String]                                $pe_console       = 'localhost',
+  Optional[Boolean]                               $disabled         = false,
+  Optional[String]                                $cron_minute      = '*/2',
+  Optional[String]                                $cron_hour        = '*',
+  Optional[String]                                $cron_weekday     = '*',
+  Optional[String]                                $cron_month       = '*',
+  Optional[String]                                $cron_monthday    = '*',
+  Optional[String]                                $log_path         = undef,
+  Optional[String]                                $confdir          = "${settings::confdir}/common_events",
+  Optional[String]                                $logfile_basepath = common_events::base_path($settings::logdir, $common_events::log_path),
+  Optional[String]                                $lockdir_basepath = common_events::base_path($settings::statedir, undef),
+  Optional[Tuple]                                 $lock_dirs        = ["${lockdir_basepath}/common_events", "${lockdir_basepath}/common_events/cache/", "${lockdir_basepath}/common_events/cache/state"],
+  Enum['DEBUG', 'INFO', 'WARN', 'ERROR', 'FATAL'] $log_level        = 'WARN',
+  Enum['NONE', 'DAILY', 'WEEKLY', 'MONTHLY']      $log_rotation     = 'NONE',
 ){
-  include common_events::install
+
+  if (
+    ($pe_token == undef)
+    and
+    ($pe_username == undef or $pe_password == undef)
+  ) {
+    $authorization_failure_message = @(MESSAGE/L)
+    Please set both 'pe_username' and 'pe_password' \
+    if you are not using a pre generated PE authorization \
+    token in the 'pe_token' parameter
+    |-MESSAGE
+    fail($authorization_failure_message)
+  }
+
+  # Account for the differences in running on Primary Server or Agent Node
+  if $facts[pe_server_version] != undef {
+    $owner              = 'pe-puppet'
+    $group              = 'pe-puppet'
+  }
+  else {
+    $owner              = 'root'
+    $group              = 'root'
+  }
+
+  unless $disabled {
+    $cron_ensure = present
+  } else {
+    $cron_ensure = absent
+  }
+
+  cron { 'collect_common_events':
+    ensure   => $cron_ensure,
+    command  => "${confdir}/collect_api_events.rb ${confdir} ${logfile_basepath}/common_events/common_events.log ${lockdir_basepath}/common_events/cache/state",
+    user     => 'root',
+    minute   => $cron_minute,
+    hour     => $cron_hour,
+    weekday  => $cron_weekday,
+    month    => $cron_month,
+    monthday => $cron_monthday,
+    require  => [
+      File["${confdir}/events_collection.yaml"],
+      File[$lock_dirs]
+    ],
+  }
+
+  file { $confdir:
```

Comment:
It looks a lot like this resource could be combined with the directories on lines 94 and 100. It would just make this file a little shorter.